### PR TITLE
Add WatsonX support

### PR DESCRIPTION
**Pull request by baptistebignaud**
It is a connector to use a LLM from WatsonX.
It requires python SDK "ibm-generative-ai"

(It might not be perfect since it is my first PR on a public repository 😄)

### CODE REVIEW

**Comment by baskaryan** [comment]
thank for @baptistebignaud, looking great! last thing is it'd be great to add a demo notebook showing how to use the class to `docs/extras/integrations/llms`

**Comment by baptistebignaud** [comment]
thanks @baskaryan for reply, I added the notebook

File: libs/langchain/langchain/llms/watsonx.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    repetition_penalty: Optional[float] = None
+    random_seed: Optional[int] = None
+    stop_sequences: Optional[List[str]] = None
+    truncate_input_tokens: Optional[int] = None
+
+    @property
+    def _llm_type(self) -> str:
+        return "watsonx"
+
+    def _call(
```
Comment (eyurtsev) [inline libs/langchain/langchain/llms/watsonx.py:36]: Could you match the method signature exactly: 

```
    def _call(
        self,
        prompt: str,
        stop: Optional[List[str]] = None,
        run_manager: Optional[CallbackManagerForLLMRun] = None,
        **kwargs: Any,
    ) -> str:
        """Run the LLM on the given prompt and input."""
```

File: libs/langchain/langchain/llms/watsonx.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+
+def _get_api_key() -> str:
+    return get_from_env("api_key", "WATSONX_API_KEY")
+
+
+class Watsonx(LLM):
+    """WatsonX LLM wrapper."""
+
+    model_name: str = "tiiuae/falcon-40b"
```
Comment (eyurtsev) [inline libs/langchain/langchain/llms/watsonx.py:18]: Would you be able to document the parameters? Especially ones that might not be obvious for interpretation?

Here's a good example of the format:

https://github.com/langchain-ai/langchain/blob/01e9d7902d4bdd0de88bc8d63c96d8d8894b8c4c/libs/langchain/langchain/llms/anthropic.py#L27-L27

File: libs/langchain/langchain/llms/watsonx.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        callbacks: Callbacks = None,
+        *,
+        tags: Optional[List[str]] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+        **kwargs: Any,
+    ) -> str:
+        try:
+            import genai
```
Comment (eyurtsev) [inline libs/langchain/langchain/llms/watsonx.py:47]: Package source: https://ibm.github.io/ibm-generative-ai/index.html 

Looks like a name collision with: https://pypi.org/project/genai/ ?

**Comment by LukaszCmielowski** [comment]
Please hold on with that connector since there is a work in progress to use official watsonx.ai library instead of experimental one `ibm-gen-ai`.

**Comment by baptistebignaud** [comment]
Great to hear that actions have been officially taken from watsonx.ai; I close the PR

**Pull request closed without merging**